src/entries/popup/pages/home/TokenDetails/KeyboardShortcutListener.tsx

```diff
@@ -0,0 +1,57 @@
+import { useCallback } from 'react';
+
+import { shortcuts } from '~/core/references/shortcuts';
+import useKeyboardAnalytics from '~/entries/popup/hooks/useKeyboardAnalytics';
+import { useKeyboardShortcut } from '~/entries/popup/hooks/useKeyboardShortcut';
+
+interface KeyboardShortcutListenerProps {
+  togglePin: () => void;
+  hideToken: () => void;
+  copyTokenAddress: () => void;
+}
+
+export function KeyboardShortcutListener({
+  togglePin,
+  hideToken,
+  copyTokenAddress,
+}: KeyboardShortcutListenerProps) {
+  const { trackShortcut } = useKeyboardAnalytics();
+
+  const condition = useCallback(() => true, []);
+
+  const handleContextMenuShortcuts = useCallback(
+    (e: KeyboardEvent) => {
+      switch (e.key) {
+        case shortcuts.tokens.PIN_ASSET.key:
+          trackShortcut({
+            key: shortcuts.tokens.PIN_ASSET.display,
+            type: 'tokens.pin',
```

Comment:
These types would be different between Tokens (Home) and Token Details. Should create a subset type for token details shortcuts